src/common/hooks/home.ts

```diff
@@ -0,0 +1,84 @@
+import { useWallet, WalletContextState } from '@solana/wallet-adapter-react';
+import { useEffect } from 'react';
+import { FeaturedProfilesQuery, useFeaturedProfilesLazyQuery } from 'src/graphql/indexerTypes';
+import { ProfilePreviewData } from '../components/elements/ProfilePreviewCard';
+
+export interface QueryContext<T> {
+  data?: T | undefined;
+  loading: boolean;
+  error?: Error | undefined;
+}
+
+//TODO remove once other profiles have enough followers to preclude this one in the backend
+const DISALLOWED_PROFILES: string[] = ['ho1aVYd4TDWCi1pMqFvboPPc3J13e4LgWkWzGJpPJty'];
+
+function useFeaturedProfilesData() {
+  const wallet: WalletContextState = useWallet();
+  const [featuredProfilesQuery, queryContext] = useFeaturedProfilesLazyQuery();
+
+  useEffect(() => {
+    async function query(): Promise<void> {
+      await featuredProfilesQuery({ variables: { limit: 25, userWallet: wallet?.publicKey } });
```

Comment:
This going to lead to excessive querying where every component using this hook is going to make a request. Please give fragments a try to allow the FeaturedProfileSection to define its data needs but still have the query executed at the page level.

Reply:
This is complementary to fragments, which I definitely want to use in addition to something like this.

We should be able to leverage apollo caching the query to avoid unnecessary triggers. However, there is probably some work to be done to get it working right.